decidim-core/app/packs/src/decidim/append_redirect_url_to_modals.js

```diff
@@ -53,23 +53,31 @@ $(() => {
   }
 
   $(document).on("click.zf.trigger", (event) => {
-    const target = `#${$(event.target).data("dialogOpen")}`;
-    const redirectUrl = $(event.target).data("redirectUrl");
-
-    if (!target || !redirectUrl) {
-      return;
+    // Try to get the <a> directly or find the closest parent <a>
+    let $target = $(event.target);
+    if (!$target.is("a")) {
+      // Find the closest parent <a> if the click is not directly on an <a>
+      $target = $target.closest("a"); 
     }
```

Comment:
@rsdeus no need to have this check. 
`closest` begins with the current element, meaning if the target is a `a` tag it will stop looking. 
https://api.jquery.com/closest/

exemple: 
```
$("a.card__link:first").closest("a")
```

gives

```
Object { 0: a.card__link, length: 1, prevObject: {…} }, 0: <a class="card__link" href="[...]">
```

```suggestion
    const $target = $(event.target).closest("a");
```

Reply:
Done ! Thanks :)